planterbox/feature.py

```diff
@@ -144,16 +146,24 @@ def run(self, result=None):
             run_hooks(module, self, result, 'before', 'feature')
             try:
                 for i, scenario in enumerate(self.scenarios):
-                    if (
-                        self.scenarios_to_run and
-                        not self.should_run_scenario(i, scenario)
-                    ):
-                        continue
+
                     (
                         self.scenario_name,
                         scenario_steps,
                         scenario_examples,
+                        scenario_tags,
                     ) = scenario
+
+                    if not matches_tag(scenario_tags, self.tag_list):
+                        continue
+
+                    if (
+                        self.scenarios_to_run and
+                        not self.should_run_scenario(i, scenario)
+                    ):
+                        continue
```

Comment:
Is this clearer if we consolidate these two separate if statements into a single conditional?